app/models/refer/referral_code.rb

```diff
@@ -7,7 +7,7 @@ class ReferralCode < ApplicationRecord
     validates :code, presence: true, uniqueness: true
 
     before_validation if: -> { Refer.code_generator } do
```

Comment:
```suggestion
    before_validation if: -> { !code? && Refer.code_generator } do
```

What about this instead since we already have a conditional here?

Reply:
I like having all the conditional logic in that same block, think it's more readable that way.